src/calculator/index.ts

```diff
@@ -185,13 +187,21 @@ export function buildLots(stockTransactions: StockTransaction[], eacHistory: EAC
         else if (sellToCoverTransaction !== undefined) {
             lots.push({
                 symbol: spaTransaction.symbol,
-                quantity: spaTransaction.quantity,
+                quantity: sellToCoverTransaction.rows.find(isSellToCoverHoldRow)?.sharesExercised || spaTransaction.quantity,
                 purchaseDate: sellToCoverTransaction.date,
-                purchasePriceUSD: sellToCoverTransaction.rows.find(isSellToCoverHoldRow)?.awardPriceUSD || 0, // TODO: This is probably not the correct purchase price. We should use the FMV price instead.
+                purchasePriceUSD: sellToCoverTransaction.rows.find(isSellToCoverSellRow)?.salePriceUSD || 0,
+            })
+        }
+        else if (holdTransaction !== undefined) {
+            lots.push({
+                symbol: spaTransaction.symbol,
+                quantity: spaTransaction.quantity,
+                purchaseDate: holdTransaction.date,
+                purchasePriceUSD: 0, // TODO: This is a missing value. We should find the FMV price instead.
             })
         }
         else {
-            throw new Error('Could not match Stock Plan Activity to Lapse or Sell To Cover transaction');
+            throw new Error('Could not match Stock Plan Activity to Lapse, Hold or Sell To Cover transaction' + JSON.stringify(spaTransaction));
```

Comment:
These errors get shown directly in the UI. Concatenating the JSON transaction to the end doesn't get displayed nicely. I think we could log the extra info in the console instead.

![Screenshot 2024-05-12 133115](https://github.com/hanzki/rsu-tax-calculator/assets/1512137/ee352654-1aae-41a0-8d37-0a1a1b49846b)